### PR TITLE
Fix input type phone

### LINKED ISSUE #310: Replace input type phone by tel
### Context

Today the input component accepts the type `phone` which is wrong, it should be `tel`.

### Description

A suggestion to avoid interface breaking changes is to add the type `tel` and redirect the type `phone` to type `tel`

### Requirements

- [x] Allow the type `tel` usage.
- [x] Don't break the interface.

### PR DESCRIPTION
**Pull request by Deivis**
## Context

Today our inputs don't have the type `tel` but have the type `phone` which isn't correct.

## Checklist
- [ ] Verify if the correct types are being received in the stories.
- [ ] Verify if the component behavior in cellphones is correct, it should pop up the phone call keyboard when focused.

## Linked Issues
- [ ] Resolves #310 

## Screenshots

Correction example in the `form` inputs stories:
Screen Shot 2019-11-25 at 15 55 54

### CODE REVIEW

File: src/Input/Input.js [hunk 1]
```diff
@@ -120,15 +120,24 @@ class Input extends React.PureComponent {
...
     const { showPassword } = this.state
-    const inputType = (type === 'password' && showPassword)
+    let inputType = (type === 'password' && showPassword)
       || multiline
       ? 'text'
       : type
 
+    inputType = inputType === 'phone'
+      ? 'phone'
+      : inputType
```
Comment (allangrds) [inline src/Input/Input.js]: Não seria melhor colocar esse ternário e o código de cima(da declaração), em uma função?

File: src/Input/Input.js [hunk 2]
```diff
@@ -120,15 +120,24 @@ class Input extends React.PureComponent {
...
     } = this.props
     const { showPassword } = this.state
-    const inputType = (type === 'password' && showPassword)
+    let inputType = (type === 'password' && showPassword)
       || multiline
       ? 'text'
       : type
 
+    inputType = inputType === 'phone'
+      ? 'phone'
```
Comment (vagnervst) [inline src/Input/Input.js]: A ideia aqui é converter o nosso `phone` para `tel`, certo? Na story de base `dark` só funcionou corretamente no Android depois que mudei esta linha para:

```suggestion
      ? 'tel'
```

File: src/Input/Input.js [hunk 3]
```diff
@@ -63,6 +63,18 @@ const validateMultiline = ({
   }
 }
 
+const getInputType = (type, showPassword, multiline) => {
```
Comment (uselessdev) [inline src/Input/Input.js]: I think those arguments would be an object, so we can use named parameters.

File: stories/Input/default.js [hunk 4]
```diff
@@ -241,7 +241,7 @@ storiesOf('Inputs', module)
           base="dark"
           mask="111-111-111"
           placeholder="Type your phone number"
-          type="text"
+          type="phone"
```
Comment (uselessdev) [inline stories/Input/default.js]: hm...

File: stories/Input/form.js [hunk 5]
```diff
@@ -265,7 +265,7 @@ storiesOf('Inputs', module)
           label="Phone number"
           mask="111-111-111"
           placeholder="Type your phone number"
-          type="text"
+          type="phone"
```
Comment (rsmelo) [inline stories/Input/form.js]: it shouldn't be tel?
```suggestion
          type="tel"
```

File: stories/Input/default.js (hunk 4 above)
Comment (rsmelo) [reply stories/Input/default.js]: it shouldn't be tel?
```suggestion
          type="tel"
```

File: stories/Input/form.js (hunk 5 above)
Comment (Deivis) [reply stories/Input/form.js]: Nope, this is one case where we are showing the change from `phone`  to `tel`.

Comment (uselessdev) [reply stories/Input/form.js]: I think its a bit weird, `tel` attribute is the standard for phone inputs in the web, why not keep as the same?

Comment (vagnervst) [reply stories/Input/form.js]: The idea is to make it retrocompatible with our old and wrong "phone" `type`, and this story here is testing if we are correctly converting `phone` to `tel` in the component's implementation.

I believe the idea is to maintain both `phone` and `tel` on our `type` prop at least for the time being :slightly_smiling_face:

**Pull request merged**